### ISSUE TITLE
Python - Migrate Configuration From Environment Variable to SSM Configuration

### ISSUE THREAD

**Issue by dealako** [communitybridge/easycla#1983]
## Summary

Python - Migrate Configuration From Environment Variable to SSM Configuration

## Background

From a Slack update to the team:

@dealako:
Ok - I made my way through all the PR’s for today. Good work everyone. We ran into an issue when @Denis Kyorov added a few additional environment variables. (not his fault :stuck_out_tongue:) We hit the lambda 4k limit on environment variable size…

One of our environment variables was an SSH key which was quite large, to begin with.

David Deal  5:36 PM
With @wanyaland help (he did 99.99% of the work), we added logic in the python side to lookup the SSM values during startup. Harold updated one key variable - the big SSH key that we are loading. It now loads via SSM instead of using the environment variable.

Summary for this configuration value only:
- Before: Serverless -> fetch SSM -> set as environment variable -> environment variable set in Lambda -> deploy lambda and env -> lambda starts -> code loads value environment variable
- After: Serverless -> deploy lambda and env (does not include this configuration value - removed from serverless) -> lambda starts -> config is called on startup -> fetch SSM value -> store value in config object -> code load values from config object

Using this approach, we:
- avoid the 4k limit on environment variables
- moved the configuration loading from the deployment to the lambda startup
- reduced the complexity of the serverless deployment
- possibly added latency to the lambda startup
- both approaches allows us to change the configuration while the lambda is running -> next time the lambda starts it will have/load any updated values

David Deal  5:38 PM
As a result of these changes, we had to adjust the CircleCI deployment to allow the python tests to pass.  While doing this I attempted to refactor/clean up the CI/CD deployment. After falling into a rabbit hole, I eventually was able to resolve the deployment problems.

## Tasks

In order to standardize how we load our configuration:
- loading of the `stage`, `AWS_REGION` via the environment is acceptable - similar to the golang side
- [ ] Identify python configuration values throughout the code - catalog these
- [ ] Identify where we load the environment variables directly (instead of using the config.py) - catalog these
- [ ] Load configuration items via SSM instead of environment variables
- [ ] Update direct environment variable usages to load the configuration from the config object which are loaded via SSM

## User Story

- As a technical lead, I want a consistent way to load configuration in the Python code

## Acceptance Criteria

The "done" criteria when this feature or problem is resolved. Such as:

1. Unit Tests added and running in CI
1. Functional Tests updated to cover feature, if applicable
1. Demonstrate the set of capabilities to the product team while the code is running in the STAGING environment.

## References

- Provide any code or specification references that would be helpful for the developer implementing this feature.

**Comment by dealako** [comment]
We've migrated at least one big configuration file to SSM.  However, we need to optimize the startup of the lambda as the SSM values would be (currently) loaded serially leading to a startup delay.  We should move the config loading to the cold start section (if similar to golang) and then have it ready/available as the warm start should be quick.

**Comment by harold** [comment]
4k of env variables 👀. That's quite a bit.

Fairly sure you meant to @ mention a different Harold - in case you do want to notify them.

Take care. (:

**Comment by dealako** [comment]
Won't fix. Out of scope of current product requirements/features.
